test/clojure-mode-syntax-test.el

```diff
@@ -89,7 +89,31 @@
         (with-clojure-buffer (concat form some-sexp)
           (end-of-buffer)
           (clojure-match-next-def)
-          (expect (looking-at "(def")))))))
+          (expect (looking-at "(def"))))))
+
+  (it "captures var name"
+    (let ((var-name "some-name"))
+      (dolist (form '("(def %s 1)"
+                      "(def %s)"
+                      "(def ^:private %s 2)"
+                      "(def ^{:private true} %s 3)"))
+        (with-clojure-buffer (format form var-name)
+          (end-of-buffer)
+          (clojure-match-next-def)
+          (cl-destructuring-bind (name-beg name-end) (match-data)
+            (expect (string= var-name (buffer-substring name-beg name-end))))))))
```

Comment:
I would much appreciate if the expectation wasn't dynamic. Otherwise it is impossible to tell a correct test from a false negative.

Would you kindly offer a follow up PR?

Reply:
Well, it's not really dynamic, though - it's always "some-name". I get your point, but I'm not sure it's worth changing this.

Reply:
You're right, I read the code a little too quickly!

Maybe `string= var-name` can be changed to `string= "some-name"` - this seems one of those cases where clarity is preferable over DRY.

One doesn't always bother learning others' variable names.

Reply:
Created a [PR](https://github.com/clojure-emacs/clojure-mode/pull/627)